lms/djangoapps/certificates/models.py

```diff
@@ -187,6 +187,34 @@ def certificate_status_for_student(student, course_id):
     return {'status': CertificateStatuses.unavailable, 'mode': GeneratedCertificate.MODES.honor}
 
 
+def certificate_info_for_user(user, course_id, grade, user_is_whitelisted=None):
+    """
+    Returns the certificate info for a user for grade report.
+    """
+    if user_is_whitelisted is None:
+        user_is_whitelisted = CertificateWhitelist.objects.filter(
+            user=user, course_id=course_id, whitelist=True
+        ).exists()
+
+    user_is_embargoed = user.profile.allow_certificate is False
+    eligible_for_certificate = not user_is_embargoed and (user_is_whitelisted or grade is not None)
```

Comment:
You can just put `user.profile.allow_certificate` in place of `not user_is_embargoed`.

And if you make it:
```
(grade is not None or user_is_whitelisted) and user.profile.allow_certificate
```
if neither of the first two are true, the user profile will not need to be queried.